packages/source-iotsitewise/src/completeDataStreams.ts

```diff
@@ -13,34 +16,93 @@ const toDataType = (propertyDataType: PropertyDataType | string | undefined): Da
+  if (alarmSource === IoTEventsSource) {
+    return data.map(({ x, y }: DataPoint): DataPoint => {
+      if (typeof y === 'string') {
+        return { x, y: parseIoTEventsAlarmData(y) };
+      }
+      return { x, y };
+    });
+  }
+
+  return data;
+};
+
 /**
  * Get completed data streams by merging together the data streams with the asset models.
  */
 export const completeDataStreams = ({
   dataStreams,
   assetModels,
+  alarms,
 }: {
   dataStreams: DataStream[];
   assetModels: Record<string, DescribeAssetModelResponse>;
-}): DataStream[] =>
-  dataStreams.map((dataStream) => {
+  alarms: AlarmsStore;
+}): DataStream[] => {
+  return dataStreams
+    .filter((dataStream) => {
+      const { assetId, propertyId } = toSiteWiseAssetProperty(dataStream.id);
+      const assetModel = assetModels[assetId];
+
+      if (assetModel == null) {
+        return false;
+      }
+
+      const alarmStateProperty: AssetModelProperty | undefined = assetModel.assetModelCompositeModels
+        ?.filter(({ type }) => type === 'AWS/ALARM')
+        .map(({ properties }) => properties).flat()
+        .find((property) => property?.id === propertyId && property?.name === 'AWS/ALARM_STATE');
+
+      if (alarmStateProperty) {
+        return Object.keys(alarms).includes(dataStream.id);
+      }
+
+      return true;
+    })
+    .map((dataStream) => {
     const { assetId, propertyId } = toSiteWiseAssetProperty(dataStream.id);
     const assetModel = assetModels[assetId];
 
-    if (assetModel == null || assetModel.assetModelProperties == null) {
-      return dataStream;
+    dataStream.associatedStreams = Object.keys(alarms)
+      .filter(id => {
+        const alarm = alarms[id];
+        return alarm.assetId === assetId && alarm.inputPropertyId === propertyId
+      })
+      .map((id) => ({
+        id,
+        type: StreamType.ALARM,
+      }))
+
+    const property = assetModel?.assetModelProperties?.find(({ id }) => id === propertyId);
+
+    if (property) {
+      return {
+        ...dataStream,
+        name: property.name,
+        unit: property.unit,
+        dataType: toDataType(property.dataType),
+      };
     }
 
-    const property = assetModel.assetModelProperties.find(({ id }) => id === propertyId);
+    const alarmStateProperty: AssetModelProperty | undefined = assetModel.assetModelCompositeModels
+      ?.filter(({ type }) => type === 'AWS/ALARM')
+      .map(({ properties }) => properties).flat()
+      .find((property) => property?.id === propertyId && property?.name === 'AWS/ALARM_STATE');
 
-    if (property == null) {
-      return dataStream;
+    if (alarmStateProperty) {
+      return {
+        ...dataStream,
+        name: alarmStateProperty.name,
+        streamType: 'ALARM',
+        data: parseAlarmData({ alarmSource: IoTEventsSource, data: dataStream.data }),
+        dataType: toDataType(alarmStateProperty.dataType),
+      }
     }
 
     return {
```

Comment:
is there a reason to spread the data stream if no other properties are appended to the object being returned